src/assignments/orbit-report.rst

```diff
@@ -0,0 +1,414 @@
+Assignment #6: Orbit Report
+===========================
+There are thousands of satellites oribiting the earth. You are tasked with creating a searchable, sortable table of satellites.
+For the purposes of this assigment, **satellite** will be defined as any objects that were purposefully placed into orbit.
+
+TODO: screen shot of finished report in the browser with numbers by each feature
+
+1. **Satellites**: Table with each row being a satellite.
+2. **Counts:** Count of all satellites and count for each type of satellite.
+3. **Search form:** Text entered limits results based on matches. Pressing enter or clicking button triggers search.
+4. **Sortable columns:** The name and type column headers can be clicked, which will sort the table using that property.
+
+
+Setup
+-----
+
+1. Create a new project on github named ``orbit-report``.
+2. Clone that repo to your computer.
+3. Create a new Angular project in the cloned repo folder using this command.
+
+   * ``$ ng new --skip-git orbit-report``
+   * It's important to include the ``--skip-git`` because we don't want Angular to create a git repo for us.
+   * Prompt answers: No Routing, Use CSS
+
+4. Commit and push your changes to github.
+
+
+Requirements
+------------
+As you accomplish each task, be sure to commit and push your changes before moving on to the next item.
+
+1) Define and Create Satellites
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+In JavaScript, TypeScript, and Angular projects you can create classes to represent entities in the project.
+For this project you need to create a class named ``Satellite`` to represent, you guessed it, a satellite. The ``Satellite``
+class needs to define the properties needed to accurately represent a satellite.
+
+1. In terminal go to the ``orbit-report`` folder
+2. Create a the class with command ``$ ng g class Satellite``
+3. Notice that the new file ``orbit-report/src/app/satellite.ts`` was created.
+4. Add these properties to the ``Satellite`` class in ``satellite.ts``:
+
+.. sourcecode:: js
+
+   name: string;
+   orbitType: string;
+   type: string;
+   operational: boolean;
+   launchDate: string;
+
+5. Add a constructor that has this method signature:
+
+.. sourcecode:: typescript
+
+   constructor(name: string, type: string, launchDate: string, orbitType: string, operational: boolean)
+
```

Comment:
Good suggestion, but I'm going to leave that detail out. They should know how constructors work by this time.

Reply:
True, but we've also trained them to follow guidance.  The instructions say "do this" and then end, so it would be natural for some students to paste in the ``constructor`` line and move on.  They would eventually realize they made an error when their code does not work, but...

Reply:
Based on your feedback I have changed this step to tell them:

- Constructor signature constructor(name: string, type: string, launchDate: string, orbitType: string, operational: boolean)
- You need to assign the class properties in the constructor.

Reply:
Better, but what does "Constructor signature constructor" mean?

Reply:
Double Constructor! In code formatting it will look better. I'm about to push up my latest fixes.

Reply:
OK.